Firestore/Example/Tests/Integration/API/FIRCountTests.mm

```diff
@@ -77,6 +109,41 @@ - (void)testCanRunCountWithOrderBys {
   XCTAssertEqual(snapshot.count, [NSNumber numberWithLong:3L]);
 }
 
+- (void)testSnapshotEquals {
+  if (![FSTIntegrationTestCase isRunningAgainstEmulator]) {
+    return;
+  }
+
+  FIRCollectionReference* testCollection = [self collectionRefWithDocuments:@{
+    @"a" : @{@"k" : @"a"},
+    @"b" : @{@"k" : @"b"},
+    @"c" : @{@"k" : @"c"}
+  }];
+
+  FIRAggregateQuerySnapshot* snapshot1 =
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"b"] count]];
+  FIRAggregateQuerySnapshot* snapshot1Same =
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"b"] count]];
+
+  FIRAggregateQuerySnapshot* snapshot2 =
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"a"] count]];
+  [self writeDocumentRef:[testCollection documentWithPath:@"d"] data:@{@"k" : @"a"}];
+  FIRAggregateQuerySnapshot* snapshot2Different =
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"a"] count]];
+
```

Comment:
Add a test case for two different `FIRAggregateQuerySnapshot` objects with the same result set but different queries. So far you've tested snapshots with the same AggregateQuery and same results, and the same AggregateQuery with different results. That third one is the only one missing to fill out the test coverage.

Reply:
Done.